contrib/build-meep.sh

```diff
@@ -0,0 +1,227 @@
+#!/bin/bash
+
+# latest version of this script can be found at:
+#   https://github.com/NanoComp/meep/blob/master/contrib/build-meep.sh
+
+# detect if script is in a directory called src/
+DESTDIR=$(pwd)
+[ ${DESTDIR##*/} = src ] && DESTDIR=$(cd $(pwd)/..; pwd)
+SRCDIR=${DESTDIR}/src
+
+cat <<EOF
+
+This sript will download or update sources, compile and install MEEP.
+Please ensure the following final paths fit your needs:
+    '${DESTDIR}/bin/meep'
+    '${DESTDIR}/share/...'
+    '${DESTDIR}/...'
+    '${SRCDIR}/<sources>'
+
+Press return to continue
+EOF
+read junk
+
+if ! lsb_release; then
+    echo "Minimum requirements:"
+    echo "    Ubuntu:"
+    echo "        sudo apt-get -y install lsb-release sudo git"
+    echo "    CentOS:"
+    echo "        sudo yum -y install redhat-lsb-core sudo git"
+    echo ""
+    exit 1
+fi
+
+set -ex
+
+ubuntu=false
+centos=false
+
+distrib=$(lsb_release -r -s)
+case "$distrib" in
+    18.04) # ubuntu 18.04 bionic
+        libpng=libpng-dev
+        libpython=libpython3-dev
+        ubuntu=true
+        ;;
+    16.04) # ubuntu 16.04 xenial
+        libpng=libpng16-dev
+        libpython=libpython3.5-dev
+        ubuntu=true
+        ;;
+    7.*) # CentOS 7.x
+        centos=true
+        ;;
+    *)
+        echo "unsupported distribution '$(lsb_release -a)', edit and fix!"
+        false
+        ;;
+esac
+
+# sudo not needed for 'make install'
+#SUDO=sudo
+SUDO=
+
+mkdir -p ${SRCDIR}
+cd ${SRCDIR}
+
+gitclone ()
+{
+    repo=${1##*/}
+    name=${repo%%.*}
+    echo $repo $name
+    if [ -d $name ]; then
+        ( cd $name; git pull; )
+    else
+        git clone --depth=1 $1
+    fi
+}
+
+autogensh ()
+{
+    sh autogen.sh PKG_CONFIG_PATH="${PKG_CONFIG_PATH}" RPATH_FLAGS="${RPATH_FLAGS}" LDFLAGS="${LDFLAGS}" CFLAGS="${CFLAGS}" CPPFLAGS="${CPPFLAGS}" \
+        --disable-static --enable-shared --prefix="${DESTDIR}" \
+        --with-libctl=${DESTDIR}/share/libctl \
```

Comment:
Maybe also need `GEN_CTL_IO=${DESTDIR}/bin/gen-ctl-io`

Reply:
Thanks for reviewing!
It is not necessary thanks to `export PATH=${DESTDIR}/bin:${PATH}`